File: 1String/1strStr/strStr.pl
#!/usr/bin/perl
#
#
use strict;

my ($source, $target) = ("source", "ce");

sub yjstrStr($$){
	my ($s, $t) = @_;
	if(undef($s) || undef($t)){
		return -1;
	}
	my @src = split("",$s);
	my @tgr = split("",$t);

	my ($slen, $tlen) = (@src, @tgr);	
	my ($i, $j) = (0, 0);
	for($i; $i < $slen-$tlen+1; $i++){
		for($j; $j < $tlen; $j++){
			if($src[$i+$j] != $tgr[$j]){
				last;
			}
			if($j == $tlen){
				return $i;
			}
		}
	}
	return -1;

}

print &yjstrStr($source, $target);

File: 1String/1strStr/strStr.py
#!/usr/local/bin/python3

#s = "source"
#t = "sourceee"

s = input("Give source:")
t = input("Give target:")

def strStr(source, target):
	if source is None or target is None:
		return -1

	for i in range(len(source) - len(target) + 1):
		for j in range(len(target)):
			if (source[i+j] != target[j]):
				break;
		else:
			return i
	return -1


print(strStr(s, t))
  
  
 
  
  
  
  
  
  
  
 
 
